### DISCUSSION TITLE
Has anyone benchmarked rust-gpu vs CUDA on comparable tasks?

### DISCUSSION THREAD

**Discussion by lachlansneff** [EmbarkStudios/rust-gpu#1028] [General]
I'm considering using rust-gpu for some GPGPU compute tasks, but I'm unsure if rust-gpu emits SPIRV code that will reach the same performance as CUDA would. Have there been any recent benchmarks on any compute-related tasks?

**Comment by charles-r-earp** [comment]
> I'm considering using rust-gpu for some GPGPU compute tasks, but I'm unsure if rust-gpu emits SPIRV code that will reach the same performance as CUDA would. Have there been any recent benchmarks on any compute-related tasks?

What kind of tasks?
